==========================================================================================================================================================

Program ini dibuat untuk memprediksi unseen data menggunakan model yang sudah dituning sebelumnya.

==========================================================================================================================================================

# Import Library

In [1]:
# data manipulation
import pandas as pd
import numpy as np

# import model
import joblib

In [2]:
# load model
with open('model.pkl', 'rb') as file_1:
    model = joblib.load(file_1)

# load metode encoding
with open('preproc.pkl', 'rb') as file_2:
    preproc = joblib.load(file_2)

# Data Loading

In [3]:
# read csv
df = pd.read_csv('digital_marketing_campaign_dataset.csv')

# mengambil 10 baris data dari dataset yang tidak digunakan sebelumnya
df = df.tail(10)

# drop kolom yang tidak ada korelasi
df_new = df.drop(columns=['AdvertisingPlatform', 'AdvertisingTool', 'CustomerID', 'Age', 'Income', 'Gender', 'CampaignChannel'])

# menampilkan dataframe
df_new

CampaignType      AdSpend  ClickThroughRate  ConversionRate  \
7990  Consideration   880.905233          0.070984        0.097500   
7991      Retention  6118.863881          0.240393        0.047079   
7992      Retention  2024.148848          0.100138        0.068403   
7993      Retention  6973.120174          0.148589        0.174582   
7994      Awareness  7875.372633          0.268637        0.082805   
7995      Awareness  8518.308575          0.243792        0.116773   
7996      Retention  1424.613446          0.236740        0.190061   
7997  Consideration  4609.534635          0.056526        0.133826   
7998  Consideration  9476.106354          0.023961        0.138386   
7999      Awareness  7743.627070          0.185670        0.057228   

      WebsiteVisits  PagesPerVisit  TimeOnSite  SocialShares  EmailOpens  \
7990             12       5.828011   10.430608            70           2   
7991             39       2.210142    6.735464            24           6   
7992             17       6.987076    5.307844            14          17   
7993             34       7.545046    6.454223             8          12   
7994              0       6.059441    9.221240            10           1   
7995             23       9.693602   14.227794            70          13   
7996             49       9.499010    3.501106            52          13   
7997             35       2.853241   14.618323            38          16   
7998             49       1.002964    3.876623            86           1   
7999             15       6.964739   12.763660             2          18   

      EmailClicks  PreviousPurchases  LoyaltyPoints  Conversion  
7990            4                  2           2848           1  
7991            5                  3           3075           1  
7992            0                  5           2733           0  
7993            1                  7           3066           0  
7994            5                  9           2448           0  
7995            6                  7            286           0  
7996            1                  5           1502           0  
7997            0                  3            738           1  
7998            5                  7           2709           1  
7999            9                  9            341           0

# Preprocessing & Predict Data

In [4]:
# melakukan preprosessing
predict_data_prep = preproc.transform(df_new)

# predict data
cek_pred = model.predict(predict_data_prep)

# menampilkan output prediction
cek_pred

array([0, 1, 1, 1, 1, 1, 1, 1, 1, 1])

In [5]:
#menggabungkan hasil prediksi ke dataframe
df_new['Hasil Prediksi'] = cek_pred
df_new['Hasil Prediksi'] = df_new['Hasil Prediksi'].astype(int)

# menampilkan dataframe
df_new

CampaignType      AdSpend  ClickThroughRate  ConversionRate  \
7990  Consideration   880.905233          0.070984        0.097500   
7991      Retention  6118.863881          0.240393        0.047079   
7992      Retention  2024.148848          0.100138        0.068403   
7993      Retention  6973.120174          0.148589        0.174582   
7994      Awareness  7875.372633          0.268637        0.082805   
7995      Awareness  8518.308575          0.243792        0.116773   
7996      Retention  1424.613446          0.236740        0.190061   
7997  Consideration  4609.534635          0.056526        0.133826   
7998  Consideration  9476.106354          0.023961        0.138386   
7999      Awareness  7743.627070          0.185670        0.057228   

      WebsiteVisits  PagesPerVisit  TimeOnSite  SocialShares  EmailOpens  \
7990             12       5.828011   10.430608            70           2   
7991             39       2.210142    6.735464            24           6   
7992             17       6.987076    5.307844            14          17   
7993             34       7.545046    6.454223             8          12   
7994              0       6.059441    9.221240            10           1   
7995             23       9.693602   14.227794            70          13   
7996             49       9.499010    3.501106            52          13   
7997             35       2.853241   14.618323            38          16   
7998             49       1.002964    3.876623            86           1   
7999             15       6.964739   12.763660             2          18   

      EmailClicks  PreviousPurchases  LoyaltyPoints  Conversion  \
7990            4                  2           2848           1   
7991            5                  3           3075           1   
7992            0                  5           2733           0   
7993            1                  7           3066           0   
7994            5                  9           2448           0   
7995            6                  7            286           0   
7996            1                  5           1502           0   
7997            0                  3            738           1   
7998            5                  7           2709           1   
7999            9                  9            341           0   

      Hasil Prediksi  
7990               0  
7991               1  
7992               1  
7993               1  
7994               1  
7995               1  
7996               1  
7997               1  
7998               1  
7999               1

# Kesimpulan

**Berdasarkan output prediksi:**

Dari 10 data yang diberikan, 9 data diprediksi akan menghasilkan konversi (1), sementara 1 data diprediksi tidak akan menghasilkan konversi (0). Hal ini menunjukkan bahwa model cenderung memprediksi konversi lebih sering, yang mungkin dikarenakan model ini melakukan prediksi berdasarkan fitur dengan tepat yang mendukung tingkat konversi lebih tinggi.

**Analisa prediksi berdasarkan fitur yang digunakan:**

1 data diprediksi tidak akan menghasilkan konversi. Faktor-faktor yang mungkin berkontribusi terhadap prediksi ini:
- Ad Spend (Pengeluaran iklan) yang lebih rendah dibanding kampanye lain.
- Click-Through Rate (CTR) dan Conversion Rate yang lebih rendah.
- Engagement metrics, seperti Website Visits dan Pages Per Visit, juga lebih rendah.

9 data diprediksi akan menghasilkan conversion. Beberapa faktor yang mungkin mempengaruhi prediksi ini adalah:
- Click-Through Rate (CTR) yang lebih tinggi.
- Conversion Rate yang relatif lebih tinggi dibandingkan kampanye lain, seperti pada kampanye ID 7993 dan 7996.
- Waktu di Situs (Time On Site) yang tinggi, menunjukkan bahwa pengunjung lebih terlibat dengan konten situs.
- Engagement metrics, seperti Social Shares, Email Opens, dan Email Clicks, yang tinggi, terutama di kampanye 7995 dan 7998.
- Loyalty Points dan Previous Purchases yang lebih tinggi, yang kemungkinan meningkatkan probabilitas konversi.